src/cmd.js

```diff
@@ -11,20 +12,21 @@ common.register('cmd', _cmd, {
   wrapOutput: true,
 });
 
-function isNullOrUndefined(val) {
-  return val === null || val === undefined;
+function isNullOrUndefinedOrEmptyString(val) {
+  return val === null || val === undefined || val === '';
 }
 
 function commandNotFound(execaResult) {
   if (process.platform === 'win32') {
     var str = 'is not recognized as an internal or external command';
-    return execaResult.code && execaResult.stderr.includes(str);
+    return execaResult.exitCode && execaResult.stderr.includes(str);
   } else {
     // On node <= 22.9.0, stdout/stderr are 'null'.
     // On node >= 22.10, stdout/stderr are 'undefined'.
-    return execaResult.code &&
-      isNullOrUndefined(execaResult.stdout) &&
-      isNullOrUndefined(execaResult.stderr);
+    return execaResult.failed &&
+      execaResult.code === 'ENOENT' &&
+      isNullOrUndefinedOrEmptyString(execaResult.stdout) &&
```

Comment:
I'd rather not include empty string in the check, because there are valid commands which exit with the empty string (ex. `false` has empty stdout, empty stderr, and code 1).

What about one of these?

```js
return execaResult.failed && execaResult.code === 'ENOENT';

// or

return execaResult.message &&
      execaResult.message.match.includes('Command failed with ENOENT');
```

Reply:
i'm good with the 1st, which i had implemented previously but since you added this utility function before the merge of this PR, i thought it implied you'd want it to be used here too so i kept it while rebasing.

Reply:
Can you rebase this on top of main branch?

Reply:
@nfischer done

Reply:
pushed. i can't vouch for win32 as i don't have a machine to run the tests locally so i didn't change anything of this part of the flow.